src/cow-react/modules/limitOrders/utils/calculateExecutionPrice.test.ts

```diff
@@ -0,0 +1,61 @@
+import { CurrencyAmount, Fraction } from '@uniswap/sdk-core'
+import { USDC_GOERLI, WETH_GOERLI } from 'utils/goerli/constants'
+import { rawToTokenAmount } from '../../../utils/rawToTokenAmount'
+import { calculateExecutionPrice, convertAmountToCurrency } from './calculateExecutionPrice'
+
+describe('calculateExecutionPrice', () => {
+  describe('When market price is less than execution price', () => {
+    const marketPrice = new Fraction(0, 1) // 0
+
+    it('When fee is zero then the result must be just output / input', () => {
+      const amount = calculateExecutionPrice({
+        inputCurrencyAmount: CurrencyAmount.fromRawAmount(USDC_GOERLI, rawToTokenAmount(6000, USDC_GOERLI.decimals)),
+        outputCurrencyAmount: CurrencyAmount.fromRawAmount(WETH_GOERLI, rawToTokenAmount(3, WETH_GOERLI.decimals)),
+        feeAmount: CurrencyAmount.fromRawAmount(USDC_GOERLI, rawToTokenAmount(0, USDC_GOERLI.decimals)),
+        marketRate: marketPrice,
+      })
+
+      // 3 / 6000 = 0.0005
+      expect(amount.toSignificant(10)).toBe('0.0005')
+    })
+
+    it('The fee must be subtracted from the execution price', () => {
+      const amount = calculateExecutionPrice({
+        inputCurrencyAmount: CurrencyAmount.fromRawAmount(USDC_GOERLI, rawToTokenAmount(6000, USDC_GOERLI.decimals)),
+        outputCurrencyAmount: CurrencyAmount.fromRawAmount(WETH_GOERLI, rawToTokenAmount(3, WETH_GOERLI.decimals)),
+        feeAmount: CurrencyAmount.fromRawAmount(USDC_GOERLI, rawToTokenAmount(250, USDC_GOERLI.decimals)),
+        marketRate: marketPrice,
+      })
+
+      // 3 / (6000 - 250) = 0.0005217391304
```

Comment:
Interesting, cause the math seems correct.

because the code is a bit verbose, makes it hard to reason about the case.
Can we add comments using GIVEN/WHEN/THEN

Also, its harder to reason if we deal with very small numbers. 
If instead of selling USDC for ETH you sell ETH for USDC, you will have simpler numbers. Also, use simpler prices like 1:1000

```js
// GIVEN: Sell 1 ETH for at least 1000 USDC   (limit price = 1000)
// GIVEN: Fee is 0.2 ETH
// GIVEN: Market Price = Limit Price = 1000
// ...
// THEN: Execution Price = 1000 / (1-0.2) = 1,250
```